python/lsst/analysis/tools/tasks/makeMetricTable.py

```diff
@@ -0,0 +1,203 @@
+        outputRefs : `lsst.pipe.base.OutputQuantizedConnection`
+        """
+
+        inputs = butlerQC.get(inputRefs)
+        skymap = inputs["skymap"]
+
+        # Make a list of tracts
+        tracts = []
+        for data in inputRefs.data:
+            tracts.append(data.dataId["tract"])
+
+        metricBundles = []
+        for inputHandle in inputs["data"]:
+            metricBundles.append(inputHandle.get())
+
+        outputs = self.run(tracts, metricBundles, skymap)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, tracts, metricBundles, skymap):
+        """Take the metric bundles and expand them out,
+        add the tract corner information and then make
+        a table of the information.
+
+        Parameters
+        ----------
+        tracts : `list`
+            A list of the tracts that the metricBundles cover
+        metricBundles : `list` of
+            `lsst.analysis.tools.interfaces._metricMeasurementBundle.MetricMeasurementBundle`
+        skymap : `lsst.skymap`
+
+        Returns
+        -------
+        tMetrics : `pipe.base.Struct` containing `astropy.table.Table`
+        """
+
+        # Make an initial dict of the columns needed
+        metricsDict = {}
+        tract = tracts[0]
+        metricsDict["tract"] = [tract]
+        corners = self.getTractCorners(skymap, tract)
+        metricsDict["corners"] = [corners]
+
+        # Add the metrics for the first tract to the dict
+        for name, metrics in metricBundles[0].items():
+            for metric in metrics:
+                fullName = f"{name}_{metric.metric_name}"
+                metricValue = metric.quantity.value
+                metricsDict[fullName] = [metricValue]
+
+        # Go through the rest of the tracts and check
+        # if any additional columns are needed and add
+        # the values to the dict
+
+        for i, metricBundle in enumerate(metricBundles[1:]):
+            tract = tracts[i + 1]
+            metricsDict["tract"].append(tract)
+            corners = self.getTractCorners(skymap, tract)
+            metricsDict["corners"].append(corners)
+            metricNames = ["tract", "corners"]
+            for name, metrics in metricBundle.items():
+                for metric in metrics:
+                    fullName = f"{name}_{metric.metric_name}"
+                    metricValue = metric.quantity.value
+                    # Check if the metric already exists in the output
+                    if fullName in metricsDict.keys():
+                        metricsDict[fullName].append(metricValue)
+                    else:
+                        values = [np.nan] * (len(metricsDict["tract"]) - 1)
+                        values.append(metricValue)
+                        metricsDict[fullName] = values
+                    metricNames.append(fullName)
+
+            # If a metric that existed for a previous tract does
+            # not exist for this one then add a nan
```

Comment:
I'd rather just skip it vs quietly making tables full of surprise nans, or at least log a warning

Reply:
I don't want any of the rows/columns to be different lengths. There needs to be something in the table cell if there is no entry.